contracts/RRUtils.sol

```diff
@@ -152,4 +152,39 @@ library RRUtils {
 
         return false;
     }
+
+    function compareNames(bytes memory self, bytes memory other) internal  returns (int){
+        bool equal = self.equals(other);
+        if(equal){ return 0; }
+
+        uint off = 0;
+        uint otheroff = 0;
+        uint head;
+        uint otherhead;
+        uint len = labelCount(self, 0);
```

Comment:
Calling this `len` is a bit confusing; it's a count of labels, not a string length.